hazelcast/src/main/java/com/hazelcast/internal/util/ServiceLoader.java

```diff
@@ -385,17 +385,22 @@ private void onNonAssignableClass(String className, Class candidate) {
             if (expectedType.isInterface()) {
                 if (ClassLoaderUtil.implementsInterfaceWithSameName(candidate, expectedType)) {
                     // this can happen in application containers - different Hazelcast JARs are loaded
-                    // by different classloaders.
+                    // by different class loaders.
                     LOGGER.fine("There appears to be a classloading conflict. "
                             + "Class " + className + " loaded by " + candidate.getClassLoader() + " implements "
                             + expectedType.getName() + " from its own class loader, but it does not implement "
                             + expectedType.getName() + " loaded by " + expectedType.getClassLoader());
                 } else {
-                    //the class does not implement interface with the expected name.
-                    LOGGER.fine("There appears to be a classloading conflict. "
-                            + "Class " + className + " loaded by " + candidate.getClassLoader() + " does not "
-                            + "implement an interface with name " + expectedType.getName() + " in both class loaders."
-                            + "the interface currently loaded by " + expectedType.getClassLoader());
+                    if (candidate.getClassLoader() != expectedType.getClassLoader()) {
+                        //the class does not implement interface with the expected name.
+                        LOGGER.fine("There appears to be a classloading conflict. "
+                                + "Class " + className + " loaded by " + candidate.getClassLoader() + " does not "
+                                + "implement an interface with name " + expectedType.getName() + " in both class loaders."
+                                + " The interface currently loaded by " + expectedType.getClassLoader());
+                    } else {
+                        LOGGER.fine("The class " + candidate.getName() + " does not implement the expected " +
+                                "interface " + expectedType.getName());
```

Comment:
Minor: the three logging statements in this method use three different conventions when referring to interfaces: "does not implement ..." vs "does not implement an interface with name ..." vs "does not implement the expected interface ..." What about unifying this?

Reply:
The text is actually unchanged from previous version, just the third case is new. I think we don't need consistency. All messages are understandable when seen in isolation... I tweaked them a bit.